bbhash_iter.go

```diff
@@ -0,0 +1,58 @@
+package bbhash
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"io"
+	"iter"
+
+	"github.com/relab/bbhash/internal/fast"
+)
+
+type BBHashSeq struct {
+	bits       []*bitVector
+	ranks      []uint64
+	reverseMap []uint64
+	hashFunc   int // Use int instead of bool incase there can be more hash functions tested than fasthash64 and SHA256. 0 = hash64, 1 = SHA256.
+	MaxLevel   int
+}
+
+// Find the chunks from slow memory
+// Currently only takes the whole sequence of input and crashes if it's not able to fit everything in the buffer.
+func ReadChunks(readerInfo io.Reader, bufSz int) iter.Seq[[]byte] {
+	return func(yield func([]byte) bool) {
+		i := 0
+		for {
+			// Create buffer and read the input into it for a certain range of bytes
+			buffer := make([]byte, bufSz)
+			_, err := readerInfo.Read(buffer)
```

Comment:
See the comment below for using `n, err` to yield a `buffer[:n]` of the expected size (for the last buffer). This should avoid special handling when using the results from the `ReadChunks` function.